From 9a7a3eb786ef4d7221c9090f2bd47933fbbdb047 Mon Sep 17 00:00:00 2001
From: Peter Crosthwaite <peter.crosthwaite@xilinx.com>
Date: Thu, 12 Jun 2014 17:05:42 -0700
Subject: [PATCH 5/7] arm64: Implement cpu_relax as yield

ARM64 has the yield instruction which has the intended semantics of
cpu_relax. Implement.

Signed-off-by: Peter Crosthwaite <peter.crosthwaite@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
---
 arch/arm64/include/asm/processor.h | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/include/asm/processor.h b/arch/arm64/include/asm/processor.h
index f9be30e..334144d 100644
--- a/arch/arm64/include/asm/processor.h
+++ b/arch/arm64/include/asm/processor.h
@@ -126,7 +126,10 @@ extern void release_thread(struct task_struct *);
 
 unsigned long get_wchan(struct task_struct *p);
 
-#define cpu_relax()			barrier()
+static inline void cpu_relax(void) {
+	barrier();
+	asm volatile("yield");
+}
 #define cpu_relax_lowlatency()                cpu_relax()
 
 /* Thread switching */
-- 
2.1.1

